pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -705,14 +717,34 @@ func (p *Pod) cleanupExcessPods(ctx context.Context, c client.Client, totalCount
 	if extraPodsCount <= 0 {
 		return nil
 	}
+	// Do not clean up more pods until observing previous operations
+	if !p.satisfiedExcessPods {
+		return errPendingOps
+	}
 
 	// Sort active pods by creation timestamp
 	sort.Slice(activePods, func(i, j int) bool {
-		return activePods[i].ObjectMeta.CreationTimestamp.Before(&activePods[j].ObjectMeta.CreationTimestamp)
+		pi := &activePods[i]
+		pj := &activePods[j]
+		iFin := slices.Contains(pi.Finalizers, PodFinalizer)
+		jFin := slices.Contains(pj.Finalizers, PodFinalizer)
+		// Keep pods that have a finalizer
+		if iFin != jFin {
+			return iFin
+		}
+		iGated := gateIndex(pi) != gateNotFound
+		jGated := gateIndex(pj) != gateNotFound
+		// keep pods that aren't gated.
+		if iGated != jGated {
+			return !iGated
+		}
+		return pi.ObjectMeta.CreationTimestamp.Before(&pj.ObjectMeta.CreationTimestamp)
```

Comment:
This would not break the tie if we have 2 excess pods, created in the same second. All pods have a finalizer, and all aren't gated. Can you confirm this is not an issue, and explain why?

Reply:
In the first iteration, any of the 2 pods will be picked. In the second iteration, if we haven't observed the finalizer removal or the pod disappearing, then will return before deleting more pods.

Reply:
I see, sounds good